nvd/nvd.go

```diff
@@ -99,39 +104,19 @@ func (u Updater) saveEntry(interval timeInterval, startIndex int) (int, error) {
 		return 0, xerrors.Errorf("unable to get url with query parameters: %w", err)
 	}
 
-	var entry Entry
-	entry, err = u.getEntry(entryURL)
+	entry, err := u.fetchEntry(entryURL)
 	if err != nil {
 		return 0, xerrors.Errorf("unable to get entry for %q: %w", entryURL, err)
 	}
-	if err = save(entry, interval.lastModEndDate); err != nil {
-		return 0, xerrors.Errorf("unable to save entry: %w", err)
-	}
-	return entry.TotalResults, nil
-}
-
-func save(entry Entry, lastModEndDate string) error {
 	for _, vuln := range entry.Vulnerabilities {
 		if err := utils.SaveCVEPerYear(filepath.Join(utils.VulnListDir(), apiDir), vuln.Cve.ID, vuln.Cve); err != nil {
-			return xerrors.Errorf("unable to write %s: %w", vuln.Cve.ID, err)
+			return 0, xerrors.Errorf("unable to write %s: %w", vuln.Cve.ID, err)
 		}
 	}
-	// we need to save LastModEndDate to avoid saving wrong time if we get errors when saving all CVEs (first run)
-	// after this we can switch to entry.Timestamp
-	timestamp, err := time.Parse(nvdTimeFormat, lastModEndDate)
-	if err != nil {
-		return xerrors.Errorf("unable to parse timestamp: %w", err)
-	}
-	// update the Last_updated.json file after saving each entry
-	// to avoid overwriting this entry if we fail to save the next entry
```

Comment:
Since we set `-eu`, the files are not committed when API calls fail in the middle.
https://github.com/aquasecurity/vuln-list-update/blob/8b9bdf21bcb9235c003a32dd91bc1f04e772803f/scripts/update.sh#L1

Therefore, we assume all requests are done successfully and update `last_updated.json` once. Please let me know if I'm missing something.

Reply:
Oh... I only tested updating `last_updated.json` file on local PC without committing.

I added this for the following case:
This is the first run, we parsed first half (as an example) of all advisories and we get error.
To avoid parsing parsed advisories, I added the update after each interval.
But we try to check all possible cases, also NVD is working on stability of its API.

This way we can only update `last_updated.json` if run is successful.

Reply:
Yes, we can assume NVD API is stable on its first run. If it doesn't work, let's go with your approach.